test/soap.test.ts

```diff
@@ -155,3 +156,87 @@ describe('convert and merge', () => {
     await conn.sobject('Lead').destroy(leadIds);
   });
 });
+
+/*------------------------------------------------------------------------*/
+
+/**
+ *
+ */
+describe('undelete', () => {
+  const contacts = [
+    {
+      LastName: 'Undelete Test #1',
+    },
+    {
+      LastName: 'Undelete Test #2',
+    },
+    {
+      LastName: 'Undelete Test #3',
+    },
+  ];
+
+  let contactIds: string[];
+
+  beforeAll(async () => {
+    // Arrange
+    const insertedContacts = await conn.sobject('Contact').create(contacts);
+    contactIds = insertedContacts.map((contact) => contact.id!);
+
+    await conn.sobject('Contact').destroy(contactIds);
+  });
+
+  /**
+   *
+   */
+  it('should undelete deleted contacts', async () => {
+    // Act
+    const res: UndeleteResult[] = (await conn.soap.undelete(
+      contactIds,
+    )) as unknown as UndeleteResult[];
```

Comment:
this type casting can be removed (if you were seeing TS errors locally you need to build jsforce first by running `npm run build:node:cjs`)